src/Microsoft.AspNetCore.Mvc.Core/Infrastructure/DefaultOutputFormatterSelector.cs

```diff
@@ -111,24 +117,33 @@ public override IOutputFormatter SelectFormatter(OutputFormatterCanWriteContext
                         contentTypes);
                 }
 
-                if (selectedFormatter == null && !_returnHttpNotAcceptable)
+                if (selectedFormatter == null)
                 {
                     _logger.NoFormatterFromNegotiation(acceptableMediaTypes);
 
-                    selectFormatterWithoutRegardingAcceptHeader = true;
+                    if (!_returnHttpNotAcceptable)
+                    {
+                        selectFormatterWithoutRegardingAcceptHeader = true;
+                    }
                 }
             }
 
             if (selectFormatterWithoutRegardingAcceptHeader)
             {
+                _logger.SelectingOutputFormatterWithoutUsingAcceptHeader();
```

Comment:
If this redundant? Like, we're going to also log @137 and @145 the same info

Reply:
Sure, will remove it.